lib/Mojolicious/Plugin/OpenAPI/Guides/OpenAPI3.pod

```diff
@@ -0,0 +1,392 @@
+
+Here are some more details about the different keys:
+
+=over 2
+
+=item * openapi, info and paths
+
+These three sections are required to make the specification valid. Check out
+L<https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md>
+for a complete reference to the specification.
+
+=item * parameters, requestBody and responses
+
+C<parameters>, C<requestBody> and C<responses> will be used to define input and output
+validtion rules, which is used by L<Mojolicious::Plugin::OpenAPI/openapi.input>
+and when rendering the response back to the client, using C<< render(openapi => ...) >>.
+
+Here OpenAPIv3 input differs from v2 spec, where C<paramters> is used for input in the
+path or query of the request. The C<requestBody> is used for input passed in the body.
+
+Have a look at L<Mojolicious::Plugin::OpenAPI/RENDERER> for more details about
+output rendering.
+
+=item * operationId and x-mojo-name
+
+See L</Route names>.
+
+=item * x-mojo-placeholder
+
+C<x-mojo-placeholder> can be used inside a parameter definition to instruct
+Mojolicious to parse a path part in a certain way. Example:
+
+  "parameters": [
+    {
+      "x-mojo-placeholder": "#",
+      "in": "path",
+      "name": "email",
+      "type": "string"
+    }
+  ]
+
+See L<Mojolicious::Guides::Routing> for more information about "standard",
+"relaxed" and "wildcard" placeholders. The default is to use the "standard"
+("/:foo") placeholder.
+
+=item * x-mojo-to
+
+The non-standard part in the spec above is "x-mojo-to". The "x-mojo-to" key
+can be either a plain string, object (hash) or an array. The string and hash
+will be passed directly to L<Mojolicious::Routes::Route/to>, while the array
+ref will be flatten. Examples:
+
+  "x-mojo-to": "pet#list"
+  $route->to("pet#list");
+
+  "x-mojo-to": {"controller": "pet", "action": "list", "foo": 123}
+  $route->to({controller => "pet", action => "list", foo => 123);
+
+  "x-mojo-to": ["pet#list", {"foo": 123}]
+  $route->to("pet#list", {foo => 123});
+
+=item * security and securitySchemes
+
+The securityScheme is added under components, where on way is to have the client
+place an apiKey in the header of the request
+
+  {
+    ...
+    "components": {
+      "securitySchemes": {
+        "apiKey": {
+          "name": "X-Api-Key",
+          "in": "header",
+          "type": "apiKey"
+        }
+      }
+    }
+  }
+
+Its then referenced under the path object as security like this
+
+  {
+    ...
+    "paths": {
+      "/pets": {
+        "get": {
+          "operationId": "getPets",
+          ...
+          "security": [
+            {
+              "apiKey": []
+            }
+          ]
+        }
+      }
+    }
+  }
+
+You can then utilize security, by adding security callback when loading
+the plugin
+
+  $self->plugin(
+    OpenAPI => {
+      spec     => $self->static->file("openapi.json")->path,
+      schema   => 'v3',
+      security => {
+        apiKey => sub ($c, $definition, $scopes, $cb) {
```

Comment:
I don't want this part to use signatures, when the rest of the documentation does not.

Reply:
fixed